maintainers/scripts/copy-tarballs.pl

```diff
@@ -180,62 +180,65 @@ sub uploadFile {
             chomp $hash;
         }
 
-        if (defined $ENV{DEBUG}) {
-            print "$url $algo $hash\n";
-            next;
-        }
-
-        if ($url !~ /^http:/ && $url !~ /^https:/ && $url !~ /^ftp:/ && $url !~ /^mirror:/) {
-            print STDERR "skipping $url (unsupported scheme)\n";
-            next;
-        }
+        for my $url (@$urls) {
+            if (defined $ENV{DEBUG}) {
+                print "$url $algo $hash\n";
+                next;
+            }
 
-        if ($isPatch) {
-            print STDERR "skipping $url (support for patches is missing)\n";
-            next;
-        }
+            if ($url !~ /^http:/ && $url !~ /^https:/ && $url !~ /^ftp:/ && $url !~ /^mirror:/) {
+                print STDERR "skipping $url (unsupported scheme)\n";
+                next;
+            }
 
-        next if defined $exclude && $url =~ /$exclude/;
+            if ($isPatch) {
+                print STDERR "skipping $url (support for patches is missing)\n";
+                next;
+            }
 
-        if (alreadyMirrored($algo, $hash)) {
-            $have++;
-            next;
-        }
+            next if defined $exclude && $url =~ /$exclude/;
 
-        my $storePath = makeFixedOutputPath(0, $algo, $hash, $name);
+            if (alreadyMirrored($algo, $hash)) {
+                $have++;
+                last;
+            }
 
-        print STDERR "mirroring $url ($storePath, $algo, $hash)...\n";
+            my $storePath = makeFixedOutputPath(0, $algo, $hash, $name);
```

Comment:
Thanks @raboof ! If we want to be pedantic, I should have moved this line (`makeFixedOutputPath`) out of the inner loop – should I do that, or is it not worth it (after all the runtime is dominated by fetching anyway)?

Reply:
Moved right before loop.